app/services/order_invoice_generator.rb

```diff
@@ -13,7 +13,7 @@ def generate_or_update_latest_invoice
         data: invoice_data
       )
     elsif comparator.can_update_latest_invoice?
-      order.invoices.last.update!(
+      order.invoices.first.update!(
```

Comment:
This relies on the knowledge of the default scope ordering by date. We could add a new method `latest` which does the same. Then we keep that knowledge within the model and are more clear when querying the "last" invoice.

Reply:
Good idea. I replaced it with `lastest`